libs/langchain-community/src/graphs/graph_document.ts

```diff
@@ -0,0 +1,69 @@
+import { Serializable } from "@langchain/core/load/serializable";
+import { Document } from "@langchain/core/documents";
+
+export class Node extends Serializable {
+  id: string | number;
+
+  type: string;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  properties: Record<string, any>;
+
+  lc_namespace = ["langchain", "graph", "document_node"];
+
+  constructor(
+    id: string | number,
+    type = "Node",
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    properties: Record<string, any> = {}
+  ) {
+    super();
+    this.id = id;
+    this.type = type;
+    this.properties = properties;
+  }
+}
+
+export class Relationship extends Serializable {
+  source: Node;
+
+  target: Node;
+
+  type: string;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  properties: Record<string, any>;
+
+  lc_namespace = ["langchain", "graph", "document_relationship"];
+
+  constructor(
+    source: Node,
+    target: Node,
+    type: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    properties: Record<string, any> = {}
+  ) {
+    super();
+    this.source = source;
+    this.target = target;
+    this.type = type;
+    this.properties = properties;
+  }
+}
+
+export class GraphDocument extends Serializable {
```

Comment:
Could this extend `Document` instead? It will be a bit more interchangeable with other RAG pipeline things in the framework if we do it that way.

Reply:
Updated - please review